### DISCUSSION TITLE
GoBilda pinpoint odometry integration

### DISCUSSION THREAD

**Discussion by rs-zip** [acmerobotics/road-runner-quickstart#453] [Q&A]
**RR Version**: 1.0.x

**Observed Behavior**: While attempting to use the *GoBilda pinpoint odometry computer* as the localizer (I modified the localizer interface to adapt this change), I faced an issue where I cannot run the feedforward and feedback to determine kS, kV, and kA (including the ramp logger process). When I initiate these op-modes, the robot moves slightly and then stops moving.

**Failed Solutions**: Since I am using the IMU computer, the robot no longer requires `inPerTick` and `trackWidth` for kinematic calculations, so I skip the process before ramp loggers. I noticed that in the `tuningOpModes` class, there is a restriction that prevents tests from running if you are not using dead wheel localizers (i.e., if `leftEncs` and `parEncs` are not empty). I attempted to add two frictional encoders without connecting the cables, which allowed the ramp loggers to become available, but the ramp regression remained blank. The reason for this seems to be that `Actions.kt` is read-only, and therefore we cannot modify the input values to use our own localizer.

I also tried determining the feedforward coefficient by connecting the encoders to motors and the control hub, then reconnecting them to the odometry computer, skipping the entire tuning process. However, a standard trajectory (like SplineTest) without the tuned localizer still does not run. I have modified the `updatePoseEstimate` method in `MecanumDrive` to accommodate the new localizer by removing the `twist` and directly using the IMU-based pose value, and the localizer has passed the localization test.

Could you kindly assist me in resolving this issue?

**Comment by rbrott** [comment]
Closing since you opened #452
